core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala

```diff
@@ -223,31 +220,26 @@ class KafkaConfigTest {
 
     //advertised listener should contain control-plane listener
     val advertisedEndpoints = serverConfig.advertisedListeners
-    assertFalse(advertisedEndpoints.filter { endpoint =>
+    assertTrue(advertisedEndpoints.exists { endpoint =>
       endpoint.securityProtocol == controlEndpoint.securityProtocol && endpoint.listenerName.value().equals(controlEndpoint.listenerName.value())
-    }.isEmpty)
+    })
 
     // interBrokerListener name should be different from control-plane listener name
     val interBrokerListenerName = serverConfig.interBrokerListenerName
     assertFalse(interBrokerListenerName.value().equals(controlEndpoint.listenerName.value()))
   }
 
   @Test
-  def testControllerListenerName() = {
-    val props = TestUtils.createBrokerConfig(0, TestUtils.MockZkConnect)
-    props.put(KafkaConfig.ListenersProp, "PLAINTEXT://localhost:0,CONTROLPLANE://localhost:4000,CONTROLLER://localhost:5000")
-    props.put(KafkaConfig.ListenerSecurityProtocolMapProp, "PLAINTEXT:PLAINTEXT,CONTROLPLANE:SSL,CONTROLLER:SASL_SSL")
-    props.put(KafkaConfig.AdvertisedListenersProp, "PLAINTEXT://localhost:0,CONTROLPLANE://localhost:4000")
-    props.put(KafkaConfig.ControlPlaneListenerNameProp, "CONTROLPLANE")
```

Comment:
This controller listener name test included control plane listener configurations, and it was declaring the config to be using ZooKeeper instead of KRaft.  We now disallow controller.listener.names when using ZooKeeper and control.plane.listener.name when using KRaft.  So now this test does what it says: it tests controller listener name, and it sets a KRaft config and leaves out the control plane listener stuff (which is tested above anyway, in `testControlPlaneListenerName()`)